File: ObfuscateConnectionData/server/spring-boot-jersey/src/main/java/com/server/reveal/DataSourceProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVDataSourceProvider;
import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.model.RVDashboardDataSource;
import com.infragistics.reveal.sdk.api.model.RVDataSourceItem;
import com.infragistics.reveal.sdk.api.model.RVSqlServerDataSource;
import com.infragistics.reveal.sdk.api.model.RVSqlServerDataSourceItem;

public class DataSourceProvider implements IRVDataSourceProvider {

    public RVDashboardDataSource changeDataSource(IRVUserContext userContext, RVDashboardDataSource dataSource) {

        if (dataSource instanceof RVSqlServerDataSource sqlDataSource){
            UpdateDataSource(sqlDataSource);
        }
        return dataSource;
    }

    public RVDataSourceItem changeDataSourceItem(IRVUserContext userContext, String dashboardsID, RVDataSourceItem dataSourceItem) {

        if (dataSourceItem instanceof RVSqlServerDataSourceItem sqlDataSourceItem)
        {
            var sqlDataSource = (RVSqlServerDataSource)sqlDataSourceItem.getDataSource();
            UpdateDataSource(sqlDataSource);

            sqlDataSourceItem.setDatabase("AutoPeople");
            sqlDataSourceItem.setSchema("dbo");
        }
        return dataSourceItem;
    }

    private static void UpdateDataSource(RVSqlServerDataSource sqlDataSource)
    {
        sqlDataSource.setHost("autosdbserver.database.windows.net");
        sqlDataSource.setDatabase("AutoPeople");
        sqlDataSource.setSchema("dbo");
    }
}

File: ObfuscateConnectionData/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs
using Reveal.Sdk;

namespace RevealSdk.Server.Reveal
{
    public class DataSourceProvider : IRVDataSourceProvider
    {
        public Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId, RVDataSourceItem dataSourceItem)
        {
            if (dataSourceItem is RVSqlServerDataSourceItem sqlDataSourceItem)
            {
                var sqlDataSource = (RVSqlServerDataSource)sqlDataSourceItem.DataSource;
                UpdateDataSource(sqlDataSource);

                sqlDataSourceItem.Database = "AutoPeople";
                sqlDataSourceItem.Schema = "dbo";
            }
            return Task.FromResult(dataSourceItem);
        }

        public Task<RVDashboardDataSource> ChangeDataSourceAsync(IRVUserContext userContext, RVDashboardDataSource dataSource)
        {
            if (dataSource is RVSqlServerDataSource sqlDataSource)
                UpdateDataSource(sqlDataSource);
            
            return Task.FromResult(dataSource);
        }

        private void UpdateDataSource(RVSqlServerDataSource sqlDataSource)
        {
            sqlDataSource.Host = "autosdbserver.database.windows.net";
            sqlDataSource.Database = "AutoPeople";
            sqlDataSource.Schema = "dbo";
        }
    }
}
